registry/remote/example_test.go

```diff
@@ -0,0 +1,542 @@
+func ExampleRepository_Fetch_manifestByDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	descriptor, err := repo.Resolve(ctx, digest) // Still need to resolve first, don't create a new descriptor with the digest, blob size is unknown
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+	pulled, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulled) || descriptor.Size != int64(len(pulled)) {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulled))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_byTag gives example snippets for downloading a manifest by tag with only one API call.
+func ExampleRepository_FetchReference_manifestByTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	reg.ManifestMediaTypes = append(reg.ManifestMediaTypes, ocispec.MediaTypeImageManifest) // Specify the expected manifest media type
+	descriptor, r, err := repo.FetchReference(ctx, tag)                                     // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err) // Handle error
+	}
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_byDigest gives example snippets for downloading a manifest by digest.
+func ExampleRepository_FetchReference_manifestByDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	descriptor, r, err := repo.FetchReference(ctx, digest) // Fetch the blob from the repository
```

Comment:
```suggestion
        descriptor, rc, err := repo.FetchReference(ctx, digest)
```

This is not against the blob API btw. Also the comment is not required since you are clearly calling FetchReference by digest. I would recommend just dropping the comment. NIT: rename reader closer to `rc`.